libraries/AP_Generator/AP_Generator_RichenPower.h

```diff
@@ -163,17 +164,41 @@ class AP_Generator_RichenPower
     // body_length as appropriate.
     void move_header_in_buffer(uint8_t initial_offset);
 
-    // servo channel used to control the generator:
-    SRV_Channel *_servo_channel;
-
     // RC input generator for pilot to specify desired generator state
     RC_Channel *_rc_channel;
 
+    // a simple heat model to avoid the motor moving to run too fast
+    // or being stopped before cooldown.
+    uint32_t last_heat_print;
+    uint32_t last_heat_update_ms;
+    float heat;
+    void update_heat();
+
+    bool generator_ok_to_run() const;
+    static constexpr float heat_required_for_run();
+
     static const uint16_t RUN_RPM = 15000;
-    static const uint16_t IDLE_RPM = 4800;
+    static const uint16_t IDLE_RPM = 6000;
+
+    static constexpr float environment_loss_factor = 0.005f;
+    // powf is not constexpr, so we create a const for it:
```

Comment:
is this comment regarding "powf" left-over from something?  I'm not sure why it's here..